rs/ledger_suite/icp/ledger/src/main.rs

```diff
@@ -801,41 +808,102 @@ fn post_upgrade(args: Option<LedgerCanisterPayload>) {
                 };
             ledger_state
         });
-    }
 
-    if ledger.ledger_version > LEDGER_VERSION {
-        panic!(
-            "Trying to downgrade from incompatible version {}. Current version is {}.",
-            ledger.ledger_version, LEDGER_VERSION
-        );
-    }
-    ledger.ledger_version = LEDGER_VERSION;
+        let upgrade_from_version = ledger.ledger_version;
+        if ledger.ledger_version > LEDGER_VERSION {
```

Comment:
In case of stable migration, we only care that the ledger state is in memory manager managed UPDATEDES_MEMORY (V2) and we check that above. For ICRC ledgers we will allow skipping versions, e.g. V2->V4. For the ICP ledger this could also be allowed (but since we do all ICP migrations, we will probably not skip versions).